### PR TITLE
[Fix] Correct the learning rate ratio in `Muon` optimizer

### LINKED ISSUE #371: Muon's Adajust LR is different from the description in the paper
## Describe the bug

In the Muon in this repository, max(1, A/B)**0.5 is used as part of lr whether "use_adjusted_lr" is True or False.

https://github.com/kozistr/pytorch_optimizer/blob/d4e7564dc58e4fa7bb0f1a1f8d9bfe5f1235116f/pytorch_optimizer/optimizer/muon.py#L205-L207

But max(1, A/B)**0.5 is not used in the paper.
And in the footnote on page 4 of the paper, there is the following:

> K. Jordan et al. 2024’s original implementation scales the updates by sqrt(max(1, A/B)), which is equivalent to our proposal (up to a global scale) if all matrices have the same second dimension; Pethick et al. 2025 and You 2025 discussed a similar issue on update scaling factors concurrently to our work.

Therefore, when "use_adjusted_lr" is True, I think max(1, A/B)**0.5 should be not used.

The paper I referenced is:

https://arxiv.org/abs/2502.16982

### LINKED ISSUE #372: Muon uses Incorrect learning rate for convolutional layers
# Describe the bug

The shape of the parameters is used to determine the learning rate in Muon. 
However, if the parameters are convolutional layers, the shape of the parameters is not in BxA format (B is the number of outputs, A is the number of inputs). 

https://github.com/kozistr/pytorch_optimizer/blob/d4e7564dc58e4fa7bb0f1a1f8d9bfe5f1235116f/pytorch_optimizer/optimizer/muon.py#L205-L207

Therefore, we need to convert the shape of the parameters to BxA format.

In addition, there are modules where the number of inputs and outputs cannot be determined from the shape of parameters alone, such as in_proj_weight in torch.nn.MultiheadAttention and weight in group convolution.

The current implementation does not work as expected for these modules.

### PR DESCRIPTION
**Pull request by kozistr**
## Problem (Why?)

Fixes #371 
Fixes #372 

## Solution (What/How?)

- [x] support `3d` and `4d` parameters correctly.
- [x] fix learning rate ratio when `use_adjusted_ratio` is `True`.

## Other changes (bug fixes, small refactors)

N/A

## Notes

N/A

### CODE REVIEW

**Comment by kozistr** [comment]
@hatonosuke I've just made a quick fix. please review this PR if you have the bandwidth. thanks!
